src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncerTest.java

```diff
@@ -216,21 +235,49 @@ public final void testBootstrapShardLeasesAtLatest()
     }
 
     /**
-     * @throws KinesisClientLibIOException
-     * @throws DependencyException
-     * @throws InvalidStateException
-     * @throws ProvisionedThroughputException
-     * @throws IOException
+     * All open and closed shards within stream's retention period should be sync'ed when lease table is empty.
      */
     @Test
-    public final void testCheckAndCreateLeasesForNewShardsAtLatest()
+    public final void testCheckAndCreateLeasesForNewShardsAtLatestWithEmptyLeaseTable()
             throws KinesisClientLibIOException, DependencyException, InvalidStateException, ProvisionedThroughputException,
             IOException {
         List<Shard> shards = constructShardListForGraphA();
         File dataFile = KinesisLocalFileDataCreator.generateTempDataFile(shards, 2, "testBootstrap1");
         dataFile.deleteOnExit();
         IKinesisProxy kinesisProxy = new KinesisLocalFileProxy(dataFile.getAbsolutePath());
 
+        shardSyncer.checkAndCreateLeasesForNewShards(kinesisProxy, leaseManager, INITIAL_POSITION_LATEST,
+                cleanupLeasesOfCompletedShards, false, shards);
+        List<KinesisClientLease> newLeases = leaseManager.listLeases();
+        Set<String> expectedLeaseShardIds = new HashSet<String>();
+        for (int i = 0; i < 11; i++) {
+            expectedLeaseShardIds.add("shardId-" + i);
+        }
+        Assert.assertEquals(expectedLeaseShardIds.size(), newLeases.size());
+        for (KinesisClientLease lease1 : newLeases) {
+            Assert.assertTrue(expectedLeaseShardIds.contains(lease1.getLeaseKey()));
+            Assert.assertEquals(ExtendedSequenceNumber.LATEST, lease1.getCheckpoint());
+        }
+        dataFile.delete();
```

Comment:
Not sure if we need this, deleteOnExit() called earlier will delete the file when the JVM terminates